SimCalorimetry/EcalSimAlgos/src/EcalLiteDTUCoder.cc

```diff
@@ -0,0 +1,241 @@
+      m_PreMix1(PreMix1)
+
+{
+  m_ebCorrNoise[0] = ebCorrNoise0;
+  assert(nullptr != m_ebCorrNoise[0]);
+  m_ebCorrNoise[1] = ebCorrNoise1;
+}
+
+EcalLiteDTUCoder::~EcalLiteDTUCoder() {}
+
+void EcalLiteDTUCoder::setFullScaleEnergy(double EBscale) {
+  //   m_maxEneEB = EBscale ;
+  m_maxEneEB = ecalPh2::maxEneEB;  //I don 't know where is setFullScaleEnergy first call
+}
+
+void EcalLiteDTUCoder::setPedestals(const EcalLiteDTUPedestalsMap* pedestals) { m_peds = pedestals; }
+
+void EcalLiteDTUCoder::setGainRatios(const EcalCATIAGainRatios* gainRatios) { m_gainRatios = gainRatios; }
+
+void EcalLiteDTUCoder::setIntercalibConstants(const EcalIntercalibConstantsMC* ical) { m_intercals = ical; }
+
+double EcalLiteDTUCoder::fullScaleEnergy(const DetId& detId) const {
+  //return detId.subdetId() == EcalBarrel ? m_maxEneEB : m_maxEneEE ;
+  return m_maxEneEB;
+}
+
+void EcalLiteDTUCoder::analogToDigital(CLHEP::HepRandomEngine* engine,
+                                    const EcalSamples& clf,
+                                    EcalDataFrame_Ph2& df) const {
+  df.setSize(clf.size());
+  encode(clf, df, engine);
+}
+
+void EcalLiteDTUCoder::encode(const EcalSamples& ecalSamples,
+                           EcalDataFrame_Ph2& df,
+                           CLHEP::HepRandomEngine* engine) const {
+  assert(nullptr != m_peds);
+
+  const unsigned int csize(ecalSamples.size());
+
+  DetId detId = ecalSamples.id();
+  double Emax = fullScaleEnergy(detId);
+
+  //....initialisation
+  if (ecalSamples[5] > 0.)
+    LogDebug("EcalLiteDTUCoder") << "Input caloSample"
+                              << "\n"
+                              << ecalSamples;
+
+  //N Gains set to 2 in the .h
+  double pedestals[ecalPh2::NGAINS];
+  double widths[ecalPh2::NGAINS];
+  //float  gains[ecalPh2::NGAINS];
+  double LSB[ecalPh2::NGAINS];
+  double trueRMS[ecalPh2::NGAINS];
+
+  double icalconst = 1.;
+  findIntercalibConstant(detId, icalconst);
+
+  for (unsigned int igain(0); igain < NGAINS; ++igain) {
+    // fill in the pedestal and width
+
+    findPedestal(detId, igain, pedestals[igain], widths[igain]);
+    //I insert an absolute value in the trueRMS
+    trueRMS[igain] = std::sqrt(std::fabs(widths[igain] * widths[igain] - 1. / 12.));
+
+    // set nominal value first
+    //findGains( detId , gains  );
+
+    LSB[igain] = Emax / (MAXADC * ecalPh2::gains[igain]);
+  }
+
+  CaloSamples noiseframe[] = {
+      CaloSamples(detId, csize),
+      CaloSamples(detId, csize),
+  };
+
+  const Noisifier* noisy[NGAINS] = {m_ebCorrNoise[0], m_ebCorrNoise[1]};
+
+  if (m_addNoise) {
+    //#warning noise generation to be checked
+    noisy[0]->noisify(noiseframe[0], engine);  // high gain
+    //if( nullptr == noisy[1] ) noisy[0]->noisify( noiseframe[1] ,
+    //                                             engine,
+    //                                              &noisy[0]->vecgau() ) ; // lwo gain
+    noisy[1]->noisify(noiseframe[1], engine);  // low gain
```

Comment:
should this be done in a loop over NGAINS?

Reply:
well it does not save lines of code... but done